### ISSUE TITLE
Fiji Train - Caffe_unet train or caffe train

### ISSUE THREAD

**Issue by BolekZapiec** [lmb-freiburg/Unet-Segmentation#18]
When starting finetuning in different Fiji instances on the same computer I get the error
ERROR Unknown command line flag 'sigint_effect' 
ERROR Unknown command line flag 'solver' 

it seems that some instance of my Fiji issue the train command as "caffe train -solver...." where as others issue "caffe_unet train -solver...." (this one doesn't work)

I'm using different instances of Fiji as it seems when starting a train, all the images that are loaded have to be used as train or validate (it would be nice if it was possible to remove images without having to close them from Fiji but I suppose this is a feature request)

Is this a version issue where the old unet plugin used "caffe_unet train" and the new one uses "caffe train" ?

**Comment by ThorstenFalk** [comment]
You can remove images from the lists by selecting them and hitting the "delete" button, I just did not add a fancy PushButton for that. More annoying is, that I did not enable multi-selection (I have to recall why). It would be nicer if you could select a bunch of list entries for shifting them around or removing them from the lists.

The caffe vs. caffe_unet issue is not version-related, caffe_unet was never meant for training, but I start believing that I should give caffe_unet all capabilities of caffe to avoid this confusion between the two binaries and a potentially installed global caffe, it really produces more headaches than it helps.

**Comment by BolekZapiec** [comment]
The delete button works?? Well I feel silly for not having tried that... thanks!

Regarding the caffe vs caffe_unet - I'm not issuing the command by hand or anything - it's the FIJI plugin that came from the update site that's issuing the caffe_unet train command. I just downgraded the plugin in the FIJI that was giving me the issue and fixed the IJPrefs.txt as you had indicated in the other issue post (it was set to caffe_unet as you assumed). I certainly never entered the caffe_unet as the caffeBinary in there which is why it made me think it was a different version that did use caffe_unet? 

I should probably post this as a separate issue as soon as I get the errors again to copy but is there a minimum on the learning rate? I notice if I go higher than 1E-4 (e.g.  5E-3 or more) I get errors and the learning stops (typically at <50 iterations). Is this the "unstable" learning we're warned about in the video?

Another question I just had that again probably belongs in a new post: aside from memory issue, what are the implications of different tile sizes. e.g. with cells that have a diameter of 20px, I have the impression finetuning goes very differently if I use tiles of 300x300 compared to 1000x1000px. I tried reading about this but so far haven't found the relevant literature - could you perhaps suggest good reading material to educate myself about this?

Thanks!

**Comment by ThorstenFalk** [comment]
I just fixed a bug in the caffe/caffe_unet detection, maybe your problem was a strange side-effect. If the error persists let me know.

If your learning rate is too high, training can diverge especially in the beginning, when no momentum is accumulated for stabilizing the gradient and ADAM has no clue on which learning rate is suitable. I guess that happened for you and in turn F1 or IoU layer were not able to compute scores due to Infs or NaNs in your scores leading to training abort. From my experience, losses above 5 indicate potential instabilities and usually lowering the learning rate in this case even leads to faster convergence.

Segmentation and Detection are only slower with smaller tiles, because more overlapping regions have to be processed multiple times. In training this is different. Larger tiles mean, that more pixels affect the gradient, this can have a regularizing effect, similar to increasing the batch size, I say similar, because the pixels are usually highly correlated which is not the case for completely different images. So training can be faster with respect to number of iterations with bigger tiles for two reasons, first more data is incorporated in the gradient per iteration, and the gradient is more stable per iteration. However wallclock-wise training will be slower with the same number of iterations, of course.

**Comment by BolekZapiec** [comment]
It seems that it is the bug you had indicated as the previous version of the plugin would keep overwriting the .unet.caffeBinary=caffe with caffe_unet each time it launched/quit Fiji. The latest version of the plugin from your ImageJ Repo has fixed this on my end. 

Thanks for solving this bug and answering my tangential questions, not to mention thank you for making this tool in the first place!
